File: 25-NumberAddition/25.js
//Given a string, add all the numbers in the string

function NumberAddition(str){ //yay regex~
  var matches = str.match(/([0-9]+)/g);
  return (matches) ? matches.reduce(function(acc,x){
    return acc + parseInt(x);
  },0) : 0;
}

function NumberAdditionNonRegex(str){
  var build = "";
  return str.split('').reduce(function(acc,x){
    if(/[0-9]/.test(x)){
      build += x;
      return acc;
    }else if(build !== ""){
      var parse = parseInt(build);
      build = "";
      return acc + parse;
    }else{
      return acc;
    }
  },0) + ((build !== "") ? parseInt(build) : 0);
}

console.log(NumberAddition("5Hello 5") === 10);
console.log(NumberAddition("123t* 1*") === 124);
console.log(NumberAddition("nothing"));

File: 25-NumberAddition/25.hs
--Given a string, add all the numbers in the string
import Data.Char

numberAddition :: String -> Int
numberAddition ""  = 0
numberAddition str@(x:xs)
  | isNum x   = (toInt $ takeWhile isNum str) + numberAddition (dropWhile isNum str)
  | otherwise = numberAddition xs
  where isNum x = elem x ['0'..'9']
        toInt xs = if xs == "" then 0 else read xs

main = do
  print $ numberAddition "5Hello 5" == 10
  print $ numberAddition "123t* 1*" == 124
  print $ numberAddition "nothing"  == 0